pkg/resources/fluentd/fluentd.go

```diff
@@ -331,6 +331,13 @@ func (r *Reconciler) reconcileDrain(ctx context.Context) (*reconcile.Result, err
 				cr.Combine(res, errors.WrapIfWithDetails(err, "removing placeholder pod for pvc", "pvc", pvc.Name))
 				continue
 			}
+
+			if r.Logging.Spec.FluentdSpec.Scaling.Drain.DeleteVolume {
+				if err := client.IgnoreNotFound(r.Client.Delete(ctx, &pvc, client.PropagationPolicy(v1.DeletePropagationBackground))); err != nil {
+					cr.CombineErr(errors.WrapIf(err, "deleting the drained PVC"))
+					continue
+				}
+			}
```

Comment:
This should be done *before* the the placeholder pod is deleted as that serves as the "guard" so that the statefulset controller cannot start its instance (pod) that would be using the PVC. Other than that, I *think* it should be fine, but someone else (who actually knows volume claims) should also think this through 😅